src/flatbuffer_types.rs

```diff
@@ -265,6 +291,47 @@ impl<'b> DbFlatBufferBuilder<'b> {
         self.builder.create_vector(sorted_runs_fbs.as_ref())
     }
 
+    fn add_uuid(&mut self, uuid: uuid::Uuid) -> WIPOffset<Uuid<'b>> {
+        let (high, low) = uuid.as_u64_pair();
+        Uuid::create(&mut self.builder, &UuidArgs { high, low })
+    }
+
+    fn time_to_unix_ts(time: &SystemTime) -> u32 {
+        time.duration_since(UNIX_EPOCH)
+            .expect("manifest expire time cannot be earlier than epoch")
+            .as_secs() as u32 // TODO: check bounds
+    }
+
+    fn maybe_time_to_unix_ts(time: Option<&SystemTime>) -> u32 {
+        time.map(Self::time_to_unix_ts).unwrap_or(0)
+    }
+
+    fn add_checkpoint(&mut self, checkpoint: &db_state::Checkpoint) -> WIPOffset<Checkpoint<'b>> {
+        let id = self.add_uuid(checkpoint.id);
+        let checkpoint_expire_time_s = Self::maybe_time_to_unix_ts(checkpoint.expire_time.as_ref());
+        let checkpoint_create_time_s = Self::time_to_unix_ts(&checkpoint.create_time);
+        Checkpoint::create(
+            &mut self.builder,
+            &CheckpointArgs {
+                id: Some(id),
+                manifest_id: checkpoint.manifest_id,
+                checkpoint_expire_time_s,
+                checkpoint_create_time_s,
+                metadata: None,
+                metadata_type: CheckpointMetadata::NONE,
```

Comment:
`manifest.fbs` does not have `metadata_type`, what is it used for?

Reply:
It's a field that gets auto-generated when you have a `union`. It specifies the type of the data stored in the union.